src/components/containers/__tests__/ConnectedContainersVisibility-test.js

```diff
@@ -74,6 +74,56 @@ describe('Basic PlotlySection rules', () => {
       it('HIDES Field', () => expect(wrapper.find('input').length).toEqual(0));
     });
 
+    describe('PlotlyPanel > PlotlySection > Field-with-no-visible-attr-based-on-customConfig', () => {
+      const wrapper = mount(
+        <TestEditor
+          {...fixtures.scatter({
+            customConfig: {
+              visibility_rules: {
+                blacklist: [{type: 'attrName', regex_match: 'color'}],
+              },
+            },
+          })}
+        >
+          <LayoutPanel>
+            <PlotlySection>
+              <Numeric attr="title.font.color" />
+            </PlotlySection>
+          </LayoutPanel>
+        </TestEditor>
+      );
+
+      it('HIDES Field based on customConfig', () =>
+        expect(wrapper.find('input').length).toEqual(0));
+      it('HIDES PlotlySection because no visible children according to custom config', () =>
+        expect(wrapper.find('div.section').length).toEqual(0));
+    });
+
+    describe('PlotlyPanel > PlotlySection > Field-with-no-visible-attr-based-on-customConfig', () => {
+      const wrapper = mount(
+        <TestEditor
+          {...fixtures.scatter({
+            customConfig: {
+              visibility_rules: {
+                blacklist: [{type: 'attrName', regex_match: 'color'}],
+              },
+            },
+          })}
+        >
+          <LayoutPanel>
+            <PlotlySection attr="title">
+              <Numeric attr="title.font.color" />
+            </PlotlySection>
+          </LayoutPanel>
+        </TestEditor>
+      );
+
+      it('HIDES Field based on customConfig', () =>
+        expect(wrapper.find('input').length).toEqual(0));
+      it('SHOWS PlotlySection if it has an attr that is accepted by customConfig', () =>
```

Comment:
shouldn't this section be hidden because it's empty?

Reply:
I could make it so, if i remove a part of the current logic that we have: https://github.com/plotly/react-chart-editor/blob/master/src/components/containers/PlotlySection.js#L51
(the || , just make it return if there's no child)

It didn't seem necessary for the current config we're trying to get right, and it's previously existing logic, that is not entirely false. (I mean it appears because we did put an attr, that is accepted by customConfig on a PlotlySection)

Reply:
to make this test easier to understand, can we make it such that there is another widget in the section with the same attr as the section?